src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -1500,20 +1488,60 @@ public GHBranchSync sync(String branch) throws IOException {
      *
      * @param org
      *            the org
-     * @return Newly forked repository that belong to you.
+     * @return Newly forked repository that belongs to you.
      * @throws IOException
      *             the io exception
+     * @deprecated Use {@link #createFork(String, String, boolean)} instead
      */
+    @Deprecated(forRemoval = true)
     public GHRepository forkTo(GHOrganization org) throws IOException {
-        root().createRequest()
-                .method("POST")
-                .with("organization", org.getLogin())
-                .withUrlPath(getApiTailUrl("forks"))
-                .send();
+        return createFork(org.getLogin(), null, false);
+    }
+
+    /**
+     * Creates a fork of this repository with optional parameters.
+     *
+     * @param organization
+     *            the organization to fork to, or null to fork to the authenticated user's account
+     * @param name
+     *            the name of the new repository, or null to use the same name as the original repository
+     * @param defaultBranchOnly
+     *            whether to fork only the default branch
+     * @return the newly forked repository
+     * @throws IOException
+     *             if an I/O error occurs
+     */
+    public GHRepository createFork(@Nullable String organization, @Nullable String name, boolean defaultBranchOnly)
+            throws IOException {
+
+        if (organization != null && organization.isEmpty()) {
+            throw new IllegalArgumentException("Organization cannot be empty");
+        }
+        if (name != null && name.isEmpty()) {
+            throw new IllegalArgumentException("Name cannot be empty");
+        }
+        if (name != null && !name.matches("^[a-zA-Z0-9._-]+$")) {
+            throw new IllegalArgumentException("Repository name contains invalid characters");
+        }
+        Requester requester = root().createRequest().method("POST").withUrlPath(getApiTailUrl("forks"));
+
+        if (organization != null) {
+            requester.with("organization", organization);
+        }
+        if (name != null) {
+            requester.with("name", name);
+        }
+        if (defaultBranchOnly) {
+            requester.with("default_branch_only", true);
+        }
+
+        requester.send();
 
         // this API is asynchronous. we need to wait for a bit
         for (int i = 0; i < 10; i++) {
-            GHRepository r = org.getRepository(name);
+            GHRepository r = organization != null
+                    ? root().getOrganization(organization).getRepository(name != null ? name : this.name)
+                    : root().getMyself().getRepository(name != null ? name : this.name);
```

Comment:
With a little copying from https://github.com/hub4j/github-api/blob/main/src/main/java/org/kohsuke/github/GHPerson.java#L123

This is approximate.  But results if far fewer API calls. 
```suggestion
            organization = organization != null ? organization : root().getMyself().getLogin();
            name = name != null ? name : this.name;
            GHRepository r;
            try {
                r =  GHRepository.read(root(), login, name);
            } catch (FileNotFoundException e) {
                r = null;
            }
```